### PR TITLE
Add support for pooling tensors without batch

### PR DESCRIPTION
**Pull request by asuhan**

### CODE REVIEW

**Comment by asuhan** [comment]
This fixes the issue I've discovered with the non-batched version of ROI align.

**Comment by dlibenzi** [comment]
You can rebase this now, and verify it is green...

**Comment by asuhan** [comment]
Test failure looks real, looking into it.

**Comment by asuhan** [comment]
@dlibenzi Fixed it, the adaptive tests passed on my machine because it was reading junk for tensors without a batch and punted to ATen implementation. I don't know how to re-trigger the build with the additional commit. PTAL, I'll then squash them together.

File: torch_xla/csrc/aten_xla_type.cpp [hunk 1]
```diff
@@ -202,8 +202,7 @@ at::Tensor AtenXlaType::__xor__(const at::Tensor& self,
 
 at::Tensor AtenXlaType::_adaptive_avg_pool2d(
     const at::Tensor& self, at::IntArrayRef output_size) const {
-  if (self.dim() != 4 ||
-      !IsSupportedAdaptiveAvgPool2d(XlaHelpers::I64List(self.sizes()),
+  if (!IsSupportedAdaptiveAvgPool2d(XlaHelpers::I64List(self.sizes()),
                                     XlaHelpers::I64List(output_size))) {
```
Comment (dlibenzi) [inline torch_xla/csrc/aten_xla_type.cpp]: Let's cache the list.

Comment (asuhan) [reply torch_xla/csrc/aten_xla_type.cpp]: Done.

**Pull request merged**